package unise;

message TopfeedDocWeightPair {
    required uint64 doc_id = 1;
    required double weight = 2;
}

message TopfeedCateWeightPair {
    required int32 cate_id = 1;
    required double weight = 2;
}

message TopfeedRequest {
    optional string log_id = 1;
    optional string cuid = 2;
    optional int64 passport_uid = 3;
    optional int64 city_id = 4;
    repeated uint64 read_docs = 5;
    repeated TopfeedDocWeightPair usercf_docs = 6;
    repeated TopfeedCateWeightPair favor_cates = 7;
}
